### DISCUSSION TITLE
Feature Request: Merge from multiple payloads

### DISCUSSION THREAD

**Discussion by regnerisch** [spatie/laravel-data#651] [Ideas]
I'm missing a merge function for merging from multiple payloads.

```php
// UserData
class UserData extends Data 
{
  public function __construct(
    public string $email,
    public ?string $name = null,
  )
}

// UpdateUserRequest
public function rules(): array
{
  return [
    'name' => 'required|string',
  ];
}

$user = User::find(1);
$data = UserData::from($user) // -> ['email' => 'max@example.com', 'name' => null]
$merged = $data->merge($updateUserRequest) // -> ['email' => 'max@example.com', 'name' => 'Max']

$user->update($merged->toArray());
```

The idea is to create a data object from e.g. a model and overwrite/extend it with e.g. a request payload. So you can easily fill the data object form an existing model and overwrite just some fields with a given request and save it.

Edit: 
The `merge` function in the given example would do the same as `from` but merges the payload.

**Comment by rubenvanassche** [comment]
Hmm, I think this can be easily added to a project with a helper function on a trait. It is not something I'll want to add to the package, data objects have already a lot of methods.

**Reply by linaspasv** [reply]
This would be very nice to have.

**Comment by linaspasv** [comment]
Maybe there is a better way but something like this worked for me...

```php
use Spatie\LaravelData\Data as BaseData;

abstract class Data extends BaseData
{
    public function merge(array | BaseData $payload): static
    {
        if ($payload instanceof BaseData) {
            $payload = $payload->toArray();
        }

        return $this->from([...$this->toArray(), ...$payload]);
    }    
}
```

**Reply by linaspasv** [reply]
Also, you can just use an available additional to achieve similar:

```php
$data->additional($newData->toArray());
```
